### PR TITLE
Added pre-commit config file

### PR DESCRIPTION
**Pull request by Raviv-S**
Signed-off-by: Raviv Schaffer <raviv.schaffer@ibm.com>

### CODE REVIEW

**Review by roytman — COMMENTED** [review]
You did not add the lint action

**Comment by Raviv-S** [comment]
What do you mean by "adding lint action"?
As I understand it, all you need for it to work is 3 files:
1. `.pre-commit-config.yaml` which I added at this PR and as golangci-lint configured to it
2. `.git/hooks/pre-commit` which is installed locally for each user when you run `pre-commit install`
3. The hook itself, in this instance it is `.golangci.yml` which I added in previous PR.

**Comment by roytman** [comment]
I mean to add automatic GitHub action which will run on each PR.
you can replace `.github/workflows/golang-security.yml` by your action. (gosec is a part of your lints)

**Comment by Raviv-S** [comment]
Added that workflow on a different PR [https://github.com/fybrik/data-movement-operator/pull/53](53#).

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    rev: v1.43.0
+    hooks:
+    - id: golangci-lint
+      name: golangci-lint
+      description: Fast linters runner for Go.
+      entry: golangci-lint run --new-from-rev=origin/master --fast
+      types: [ go ]
+      language: golang
+      pass_filenames: false
+      verbose: true
```
Comment (roytman) [inline .pre-commit-config.yaml:12]: pls add a new line

**Review by roytman — APPROVED** [review]

**Pull request merged**